subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/concurrent/RessurectingThread.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl.concurrent
+
+import kotlin.concurrent.thread
+
+
+/**
+ * A thread that comes back to life when [poke]d.
+ */
+internal
+class RessurectingThread(val name: String, val block: () -> Unit) {
+
+    private
+    var thread: Thread? = null
+
+    /**
+     * Checks if the thread is alive, ressurecting it if it's not.
+     */
+    fun poke() = synchronized(this) {
+        if (thread?.isAlive != true) {
+            thread = thread(name = name, block = block)
```

Comment:
Do you want this to be a daemon thread so it dies when the main thread dies?

Reply:
Of course!